hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -236,8 +238,8 @@ protected HoodieWriteMetadata<O> logCompact(String logCompactionInstantTime, boo
     WriteMarkersFactory.get(config.getMarkersType(), table, logCompactionInstantTime);
     HoodieWriteMetadata<T> writeMetadata = table.logCompact(context, logCompactionInstantTime);
     HoodieWriteMetadata<O> logCompactionMetadata = convertToOutputMetadata(writeMetadata);
-    if (shouldComplete && logCompactionMetadata.getCommitMetadata().isPresent()) {
-      completeLogCompaction(logCompactionMetadata.getCommitMetadata().get(), table, logCompactionInstantTime);
+    if (shouldComplete) {
+      commitLogCompaction(logCompactionInstantTime, logCompactionMetadata, Option.of(table));
```

Comment:
And is this the bug fixed?

Reply:
nope. we are just expecting HoodieWriteMetadata in commitLogCompaction method